src/main/java/com/exactpro/th2/mstore/MessageProcessor.java

```diff
@@ -212,11 +212,17 @@ private static void reject(Confirmation confirmation) {
 
     private GroupedMessageBatchToStore toCradleBatch(String group, List<RawMessage> messagesList) throws CradleStorageException {
         GroupedMessageBatchToStore batch = cradleStorage.getEntitiesFactory().groupedMessageBatch(group);
-        for (RawMessage message : messagesList) {
-            MessageToStore messageToStore = ProtoUtil.toCradleMessage(message);
-            batch.addMessage(messageToStore);
+        try {
+            for (RawMessage message : messagesList) {
+                MessageToStore messageToStore = ProtoUtil.toCradleMessage(message);
+                batch.addMessage(messageToStore);
+            }
+            return batch;
+        } catch (CradleStorageException e) {
+            invalidateCacheForRawMessages(messagesList);
```

Comment:
toCradleBatch should not be responsivle for cache maintenace.
Invalid messages should not be put in cache at first place